pkg/apis/topology/v1alpha1/types_partition.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Partition defines the selection of a set of shards along multiple dimensions.
+//
+// Partitions can get automatically generated through a partitioner or manually crafted.
+//
+// +crd
```

Comment:
global: comment like this:


```
// all tags in one group

// *after a blank line* -> godoc
type Partition struct
```

for types & fields - the tags will still work and not end up in GoDoc

Reply:
I followed what has been done pkg/apis/apis/v1alpha1/types_apiexport.go but will adopt your approach. I will only keep +optional as it is in my opinion valuable information.

Reply:
All tags need to go in a separate comment section. If the field is optional, the tags will mark it so in structured documentation. If you furthermore want to describe it as optional in the GoDoc, that's ok, but let's do so with a full English sentence instead of asking users to become robots and read tags :)